src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -39,50 +36,42 @@ def acr_import(cmd,
         raise CLIError(INVALID_SOURCE_IMAGE)
     source_image = source
 
-    slash = source.find('/')
-
     ImportImageParameters, ImportSource, ImportMode = cmd.get_models(
         'ImportImageParameters', 'ImportSource', 'ImportMode')
-    if slash < 0:
-        if not source_registry:
-            from knack.prompting import prompt, NoTTYException
-            try:
-                source_registry = prompt(SOURCE_REGISTRY_MISSING)
-            except NoTTYException:
-                raise CLIError(NO_TTY_ERROR)
-        if not is_valid_resource_id(source_registry):
+
+    if source_registry:
+        if is_valid_resource_id(source_registry):
+            source = ImportSource(resource_id=source_registry, source_image=source_image)
+        else:
             registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry, source_registry)
             if registry:
-                source_registry = registry.id
-        source = ImportSource(resource_id=source_registry, source_image=source_image)
+                source = ImportSource(resource_id=registry.id, source_image=source_image)
+            else:
+                raise CLIError(SOURCE_REGISTRY_NOT_FOUND)
     else:
-        source_registry_login_server = source_image[:slash]
-        source_image = source_image[slash + 1:]
-        if not source_image or not source_registry_login_server:
-            raise CLIError(INVALID_SOURCE_IMAGE)
-        registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry_login_server)
-        if registry:
-            if source_registry and \
-               source_registry.lower() != registry.id.lower() and \
-               source_registry.lower() != registry.name.lower() and \
-               source_registry.lower() != registry.login_server.lower():
-                raise CLIError(REGISTRY_MISMATCH)
-            source = ImportSource(resource_id=registry.id,
-                                  source_image=source_image)
-        else:
+        slash = source_image.find('/')
+        if slash > 0:
+            registry_uri = source_image[:slash]
+            source_image = source_image[slash + 1:]
             if source_registry_password:
```

Comment:
If only user name is provided, it is not going to authenticate by any means.

Reply:
I see the current code path silently ignores this case, which is less preferred since the behavior could be unexpected to users.

If providing `--username` without `--password` is not a valid scenario, we should just error out similar to
https://github.com/Azure/azure-cli/blob/466cc22717bef747deb63768ccae40a8e2ef5325/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py#L154

Reply:
If username is specified without password, an error throws as unauthorized request.

Reply:
Yes, that's what I was saying, the client can show a message, see the above link as an example.